src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3945,6 +3945,32 @@ GenTree* Lowering::TryLowerAndOpToAndNot(GenTreeOp* andNode)
     return andnNode;
 }
 
+//----------------------------------------------------------------------------------------------
+// Lowering::LowerBswapOp: Tries to contain GT_BSWAP node when possible
+//
+// Arguments:
+//    node - GT_BSWAP node to contain
+//
+// Notes:
+//    Containment is not performed when optimizations are disabled
+//    or when MOVBE instruction set is not found
+//
+void Lowering::LowerBswapOp(GenTreeOp* node)
+{
+    assert(node->OperIs(GT_BSWAP, GT_BSWAP16));
+
+    if (!comp->opts.OptimizationEnabled() || !comp->compOpportunisticallyDependsOn(InstructionSet_MOVBE))
+    {
+        return;
+    }
+
+    GenTree* operand = node->gtGetOp1();
+    if (node->TypeGet() == operand->TypeGet() && IsContainableMemoryOp(operand) && IsSafeToContainMem(node, operand))
```

Comment:
```suggestion
    GenTree* operand = node->gtGetOp1();
    unsigned swapSize = node->OperIs(GT_BSWAP16) ? 2 : genTypeSize(node);
    if ((genTypeSize(operand) == swapSize) && IsContainableMemoryOp(operand) && IsSafeToContainMem(node, operand))
```

The JIT does not support many operations beyond stores and loads on small integer types. In particular `BSWAP16` nodes in the JIT always produce and work on `TYP_INT` values, so the existing check is not right for `BSWAP16`.

Reply:
Can we actually just make the `BSWAP16` a `TYP_SHORT` instead since we are now generating the zero extension inside the code gen? At least I'm not seeing any tests failing due to it. Also because its currently `TYP_INT` its guarding against that we don't end up containing loads for small indirections so there's no need to zero extend on the `INS_movbe` path.

Reply:
> Can we actually just make the BSWAP16 a TYP_SHORT instead since we are now generating the zero extension inside the code gen? 

This would be quite unusual in the JIT's IR, there are no small-typed arithmetic nodes. I would avoid changing the semantics of `GT_BSWAP16` in this PR.

> Also because its currently TYP_INT its guarding against that we don't end up containing loads for small indirections so there's no need to zero extend on the INS_movbe path.

Only because this means that the optimization never applies to 16-bit bswaps. It will be necessary once that is fixed.

Reply:
> This would be quite unusual in the JIT's IR, there are no small-typed arithmetic nodes. I would avoid changing the semantics of GT_BSWAP16 in this PR.

Hmm, makes sense. Looking at the history it was the case originally.

> Only because this means that the optimization never applies to 16-bit bswaps. It will be necessary once that is fixed.

Oh, right, oops. Not sure how did I not notice that.